test/common/http/utility_test.cc

```diff
@@ -1289,6 +1289,9 @@ TEST(Url, ParsingTest) {
   // Test url with non-default ports.
   validateUrl("https://www.host.com:8443", "https", "www.host.com:8443", "/", 8443);
   validateUrl("http://www.host.com:8080", "http", "www.host.com:8080", "/", 8080);
+
+  // Make sure we survive parsing unicode URL.
+  validateUrl("https://\xe5\x85\x89.example/", "https", "xn--54q.example", "/", 443);
```

Comment:
So what was the prior behavior?  Would all unicode URLs be deemed invalid?  If so I think this merits guarding in case upstreams can't handle unicode and the change in Envoy validation causes problems.  If some unicode URLS passed through previously maybe it doesn't make enough difference to merit the guard.  WDYT?

Either way probably worth a release note.

Reply:
> Would all unicode URLs be deemed invalid?

Unicode host name segments of URI's are invalid, at the transport level.

Unicode -> PunyCode. The resulting URI is a valid hostname by the DNS spec.

Reply:
@dio In what circumstances we would get unicode URL in non-punycode format? I'm confused why we need handling that.

Reply:
Sorry for the confusion. This is motivated by porting the test case from https://quiche.googlesource.com/googleurl/+/refs/heads/master/test/basic_test.cc#31.

I think It is possible if we want to process input from config or logging. However, if this is not a valid use case. I can skip checking for this one.